cpu/cortexm_common/vectors_cortexm.c

```diff
@@ -79,6 +79,9 @@ void reset_handler_default(void)
         *(dst++) = STACK_CANARY_WORD;
     }
 
+    /* Workaround for compiling without optimization: -O0 */
```

Comment:
Suggestion for a longer explanation: "Any local stack-allocated variables used in this function will be overwritten in the above loop. When compiling with -O0 src may be allocated on the stack, leading to a hard fault when dereferencing it. The workaround is to assign src after overwriting the stack."